### PR TITLE
Foundations Slide Deck Content & Notes Update

### PR DESCRIPTION
**Pull request by jordanmccullough**
Tidying up some notes and some command examples.

### CODE REVIEW

File: presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+chapter: Log
+layout: slide
+tags: ['log']
+---
+
+	git log -S<string-in-patch>
```
Comment (brntbeer) [inline presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md]: Currently already lives in [log additional options](https://github.com/github/teach.github.com/blob/gh-pages/presentations/_posts/foundations/log/0005-01-01-Additional-Options.md)

Recommend moving this first, also don't forget to put a space: `-S <string-in-patch>` vs `-S<string-in-patch>`

Comment (matthewmccullough) [inline presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md]: @brntbeer To your excellent point, I'd like to see it killed (too many things on one slide) on the `log additional options` page

File: presentations/_posts/foundations/stash/0002-01-01-Stash-Commands.md [hunk 2]
```diff
@@ -7,22 +7,10 @@ categories: ['slidecontent']
...
-	git stash pop <name>
-
-	# Remove a stash
-	git stash drop <name>
-
-	# Merge & keep the latest stash
-	git stash apply
-
-	# Drop all stashes
-	git stash clear
```
Comment (matthewmccullough) [inline presentations/_posts/foundations/stash/0002-01-01-Stash-Commands.md:28]: Super happy at this segmenting of stash. I do like to reveal it in pieces like this.

File: presentations/_posts/foundations/stash/0003-01-01-Stash-Commands.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+	# Merge & delete the latest stash
+	git stash pop
+
+	# Merge & delete a stash
+	git stash pop <name>
+
+	# Merge & keep the latest stash
+	git stash apply
```
Comment (matthewmccullough) [inline presentations/_posts/foundations/stash/0003-01-01-Stash-Commands.md:16]: I like pop and apply being together on the same slide, as I love to contrast and compare them.

File: presentations/_posts/foundations/stash/0005-01-01-Branch-from-a-Stash.md [hunk 4]
```diff
@@ -5,4 +5,4 @@ tags: ['stash']
 categories: ['slidecontent']
 ---
 
-	$ git branch <branchname> stash@{0}
\ No newline at end of file
+	$ git branch <branchname> <stash-name>
```
Comment (matthewmccullough) [inline presentations/_posts/foundations/stash/0005-01-01-Branch-from-a-Stash.md:8]: Probably just needs some speaker notes about how the format is `stash@{NUM}`

File: presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md (hunk 1 above)
Comment (jordanmccullough) [inline presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md]: @brntbeer Thanks for the note. Oddly, Git plays nice with or without the space between the `-S` and the string. However, for clarity's sake the space definitely makes sense.

**Comment by jordanmccullough** [comment]
Considerably updated slides, fine tuned some diagrams that needed work, added a new build-out for the Commit Structure (commit-->tree-->blob) graphic.

This is near-ready for a merge, if I do say so myself.

File: presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md (hunk 1 above)
Comment (brntbeer) [inline presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md]: Ah. though this is what tripped us up a few months ago when we were teaching down in San jose area together!

**Comment by matthewmccullough** [comment]
A quick review says good to go.  What say you @brntbeer since you gave this detailed eyes on the first pass?

**Comment by brntbeer** [comment]
ship it. if there's issues we'll fix them as we run through them

**Comment by jordanmccullough** [comment]
:ship:ed. Thanks for the review guys.

**Pull request merged**